mountinfo/mounted_linux_test.go

```diff
@@ -324,34 +317,43 @@ func TestMountedBy(t *testing.T) {
 
 			// Check the public MountedFast() function as a whole.
 			mounted, sure, err := MountedFast(m)
-			if err == nil {
-				if !openat2Supported {
-					// stat does not detect bind-mounts correctly.
-					if tc.isBind && mounted == true {
-						t.Errorf("MountFast: expected mounted as false for bind-mounts, got %v", mounted)
-					}
-					if !tc.isBind && mounted != exp {
+
+			// for any error, regardless of kernel version, sure and mounted is false.
+			if err != nil {
+				if sure == true {
+					t.Errorf("MountFast: expected sure to be false, got %v", sure)
+				}
+				if mounted != false {
```

Comment:
The code would be easier to read if you use the same comparison (either `== false` or `!= true`) in both cases.